docs/arch/adr-0001-style-guide.md

```diff
@@ -0,0 +1,76 @@
+# Style Guide
+
+
+## Context
+
+We need a style guide for our actions repo. This structure provides the following benefits:
+- developers can spend more time on solving the problem
+- PR reviews are limited to discussing intent
+- format related feedback can point to a collective style guide instead of personal preferences
+- consumers know how to find what they're looking for
+- developers know where to look when fixing something
+
+## Options
+
+This style guide is a collection of individual options. Instead of providing permutations of these options,
+this ADR provides a section for an initial proposal for several style guide topics and a section
+that contains the agreed upon formats. This ADR should be updated to remove reference to the initial
+proposal prior to merging in order to avoid confusion.
+
+As an alternative, each topic could be its own ADR. Reviewers may request this approach, which will require
+breaking this ADR into several ADRs and associated PRs.
+
+## Proposal
+
+### Actions
+- actions live at the repo root so that usage looks like `uses: dbt-labs/actions/<action-name>@<tag>`
+- actions live in directories with the format `<platform/tool>/<resource>/<action>`, e.g.
+    - `./hatch/environment/create`
+    - `./pypi/release/view`
+- actions have a generic scope with a focused context to support reusability
+- actions perform a single task on a single object within the context of a platform/tool
+
+### Workflows
+- workflows need to live in `.github/workflows`
+- we prefer actions to workflows
+    - actions can be inserted into other actions and as part of workflows
+    - actions can inherit environment variables
+
+### Formatting
+- job ids, step ids, and variables are in kebab case
+- job names, step names, and description fields follow dbt docs standards (capitalize first word only)
+- extra descriptors should be avoided unless required for disambiguation, e.g.
+    - `version-number` -> `version`
+    - `archive-name` -> `archive`
+- yaml files use a four space tab
+- scripts use environment variables in `env` instead of inline substitution like `${{ inputs.value }}`
```

Comment:
We should have a scripts section.  When should we write a script vs just doing it inline?  Where should scripts live? etc

Reply:
Yeah, that probably makes sense. To initially respond to the few examples you gave:
- I am moving more and more towards scripts always; so basically if it's more than a line, make it a script
- scripts should live alongside the `action.yml` which calls them, multiple actions should not depend on the same script